src/components/ContributionsEpic.tsx

```diff
@@ -4,11 +4,21 @@ import { body, headline } from '@guardian/src-foundations/typography';
 import { palette } from '@guardian/src-foundations';
 import { space } from '@guardian/src-foundations';
 import { PrimaryButton } from './PrimaryButton';
+import { getCountryName, getLocalCurrencySymbol } from '../lib/geolocation';
 
-const currencySymbol = '£';
+const countryCode = 'GB';
 
-const replacePlaceholders = (content: string): string =>
-    content.replace(/%%CURRENCY_SYMBOL%%/g, currencySymbol);
+const replacePlaceholders = (content: string, countryCode?: string): string => {
```

Comment:
We spoke offline about potentially refactoring the `replacePlaceholders` function to abstract the actual string replacement logic into a separate function that takes the placeholder name and the the value to replace it with. That would probably keep the logic in this function a bit cleaner and easier to read, focused on the logic to retrieve the data for each placeholder, separate from the "verbosity" of actually replacing the string.

Reply:
Cool thanks for summarising, agree with your points here :+1: Fixed this d2aa1ad.

Reply:
Looks great 🏅

Reply:
Is this `replace` equivalent to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace ?

Reply:
Also, what does this function do if `replacement?` is falsy? `''` (empty string) seems like a sensible replacement option but the function will fail on that. Maybe `replacement ?? false` might work now - 'nullish coalescing' is the new name for this apparently. More generally, when would it make sense to not provide a replacement? (I can't think of this case at least.)

Reply:
@nicl - possible I've missed something here but I think Tom's logic is appropriate. In the context of the `replacePlaceholders` function, it makes sense for the `replace` function not to perform the replacement unless there's some data for the placeholder to be replaced with. Returning the original "unreplaced" content (with the placeholder untouched) seems like a good fallback. We can also add extra checks around this to ensure no placeholder text is ever served to the client - but I wouldn't think this would be done here (perhaps closer to the targeting logic?).

In any case, I can see us moving away from string replacement at some point to adopt a more "template rendering" like approach.

Reply:
Possibly I've misunderstood, but if `replacement` is falsy, the function is a no-op right? In which, case why allow this parameter to be optional?

Also, again, I'm unclear why we're not just using the standard lib `replace` function?

Reply:
I've refactored the replacement logic to just use standard string replacement.

Reply:
Hi @nicl @andre1050 I know this has been merged already but just wanted to chip in to explain the logic here. The `replace` function I added was just a thin wrapper around `String.prototype.replace` which added two things: 1) build up the regex and 2) don't do the replacement if the replacement value is falsy. This was to preserve the behaviour we have in frontend where we won't ever replace a placeholder with nothing. Instead we'll leave it if we don't have a replacement value, and then halt the showing of the epic if there are unreplaced placeholders. I think I definitely could have explained this better and picked a more meaningful name for `replace` to better convey this! Sorry about that.

> Possibly I've misunderstood, but if replacement is falsy, the function is a no-op right? In which, case why allow this parameter to be optional?

That's correct, if the replacement is falsy then it's a no-op. The parameter is optional to handle the fact that the replacement value can in some cases be `undefined`.

It sounds like you got to a place where you were happy to merge, so thanks for following through with this while I was out!